docs/guides/connecting_to_your_data/filesystem/pandas.md

```diff
@@ -1,136 +1,131 @@
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_python_example.py#L36
 ```
 
 </TabItem>
 </Tabs>
 
-### 3. Write a `BatchRequest`
+If you specified a directory containing CSV files you will see them listed as `Available data_asset_names` in the output.
+
+Feel free to adjust your configuration and re-run `test_yaml_config` as needed.
 
-In a `RuntimeBatchRequest` the `data_asset_name` can be any unique name to identify this batch of data. Please update the `data_asset_name` to something meaningful to you.
+### 4. Save the Datasource configuration to your DataContext
+
+Save the configuration into your `DataContext` by using the `add_datasource()` function.
 
 <Tabs
-  defaultValue="csv"
+  groupId="yaml-or-python"
+  defaultValue='yaml'
   values={[
-    {label: 'CSV', value: 'csv'},
-    {label: 'CSV with custom delimiters', value: 'csv_delimiters'},
-    {label: 'CSV without a header', value: 'csv_headerless'},
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
   ]}>
-  <TabItem value="csv">
-
-Add the path to your csv on a filesystem to the `path` key under `runtime_parameters` and run the following code:
+  <TabItem value="yaml">
 
-```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas.py#L24-L31
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L37
 ```
 
-  </TabItem>
-  <TabItem value="csv_delimiters">
-
-  **TODO** do this
-
-  </TabItem>
-  <TabItem value="csv_headerless">
+</TabItem>
+<TabItem value="python">
 
-  **TODO** do this
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_python_example.py#L38
+```
 
-  </TabItem>
+</TabItem>
 </Tabs>
 
-### 4. Test your datasource configuration by getting a `Batch`
+### 5. Test your new Datasource
 
-Load a `Batch` of your data to test if your `Datasource` configuration works by running the following code:
+Verify your new Datasource by loading data from it into a `Validator` using a `BatchRequest`.
 
-```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas.py#L39
-```
-
-Congratulations! If no errors are shown here, you've just connected to your data on a filesystem using pandas!
+Add the path to your CSV in the `path` key under `runtime_parameters`.
 
-:::tip Next steps
-After connecting to data, usually people validate data!
-**TODO** insert link or have this be an include.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L39-L45
+```
+Then load data into the `Validator`.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L53-L59
+```
 
-Right now a Batch is just Data + Metadata that is recognized by GE. If you want to actually Validate your data with Expectations, you will need a Batch + ExpectationSuite.
-:::
+<Congratulations />
```

Comment:
Pandas-like how-to-guides will support both a `RuntimeBatchRequest` with path (using a `RuntimeDataConnector`) and a `BatchRequest` with `data_asset_name` (using an `InferredAssetDataConnector`). This will be submitted as a separate PR as discussed during review